test/adaptive_load/fake_plugins/fake_step_controller/fake_step_controller_test.cc

```diff
@@ -0,0 +1,187 @@
+  Envoy::ProtobufWkt::Any config_any;
+  config_any.PackFrom(config);
+  CommandLineOptions options;
+  auto& config_factory =
+      Envoy::Config::Utility::getAndCheckFactoryByName<StepControllerConfigFactory>(
+          "nighthawk.fake-step-controller");
+  EXPECT_EQ(config_factory.name(), "nighthawk.fake-step-controller");
+}
+
+TEST(FakeStepControllerConfigFactory, CreateStepControllerCreatesCorrectPluginType) {
+  FakeStepControllerConfig config;
+  Envoy::ProtobufWkt::Any config_any;
+  config_any.PackFrom(config);
+  CommandLineOptions options;
+  auto& config_factory =
+      Envoy::Config::Utility::getAndCheckFactoryByName<StepControllerConfigFactory>(
+          "nighthawk.fake-step-controller");
+  StepControllerPtr plugin = config_factory.createStepController(config_any, options);
+  EXPECT_NE(dynamic_cast<FakeStepController*>(plugin.get()), nullptr);
+}
+
+TEST(FakeStepControllerConfigFactory, ValidateConfigWithBadConfigProtoReturnsError) {
+  Envoy::ProtobufWkt::Any empty_any;
+  auto& config_factory =
+      Envoy::Config::Utility::getAndCheckFactoryByName<StepControllerConfigFactory>(
+          "nighthawk.fake-step-controller");
+  absl::Status status = config_factory.ValidateConfig(empty_any);
+  EXPECT_THAT(status.message(), HasSubstr("Failed to parse"));
+}
+
+TEST(FakeStepControllerConfigFactory, ValidateConfigWithWellFormedIllegalConfigReturnsError) {
+  FakeStepControllerConfig config;
+  // Negative value fails config validation:
+  config.set_fixed_rps_value(-1);
+  Envoy::ProtobufWkt::Any config_any;
+  config_any.PackFrom(config);
+  auto& config_factory =
+      Envoy::Config::Utility::getAndCheckFactoryByName<StepControllerConfigFactory>(
+          "nighthawk.fake-step-controller");
+  absl::Status status = config_factory.ValidateConfig(config_any);
+  EXPECT_THAT(status.message(), HasSubstr("Negative fixed_rps_value"));
+}
+
+TEST(FakeStepControllerConfigFactory, ValidateConfigWithDefaultConfigReturnsOk) {
+  FakeStepControllerConfig config;
+  Envoy::ProtobufWkt::Any config_any;
+  config_any.PackFrom(config);
+  auto& config_factory =
+      Envoy::Config::Utility::getAndCheckFactoryByName<StepControllerConfigFactory>(
+          "nighthawk.fake-step-controller");
+  absl::Status status = config_factory.ValidateConfig(config_any);
+  EXPECT_TRUE(status.ok());
+}
+
+TEST(FakeStepControllerConfigFactory, ValidateConfigWithValidConfigReturnsOk) {
+  FakeStepControllerConfig config;
+  config.set_fixed_rps_value(1);
+  Envoy::ProtobufWkt::Any config_any;
+  config_any.PackFrom(config);
+  auto& config_factory =
+      Envoy::Config::Utility::getAndCheckFactoryByName<StepControllerConfigFactory>(
+          "nighthawk.fake-step-controller");
+  absl::Status status = config_factory.ValidateConfig(config_any);
+  EXPECT_TRUE(status.ok());
+}
+
+TEST(FakeStepController, GetCurrentCommandLineOptionsReturnsRpsFromConfig) {
+  FakeStepControllerConfig config;
+  config.set_fixed_rps_value(5678);
```

Comment:
Maybe create a shared constant for the magic number?

Reply:
Done -- keeping these constants local since they have no shared meaning across tests